test/csuite/wt8057_compact_stress/main.c

```diff
@@ -0,0 +1,381 @@
+        remove_records(session, uri1, key_range_start, key_range_start + NUM_RECORDS / 3);
+        remove_records(session, uri2, key_range_start, key_range_start + NUM_RECORDS / 3);
+
+        file_sz_before = get_file_size(session, uri1);
+
+        testutil_check(session->compact(session, uri1, NULL));
+
+        file_sz_after = get_file_size(session, uri1);
+
+        /* Check if there's at least 10% compaction. */
+        printf(" - Compressed file size MB: %f\n - Original file size MB: %f\n",
+          file_sz_after / (1024.0 * 1024), file_sz_before / (1024.0 * 1024));
+
+        /* Put the deleted records back. */
+        populate(session, key_range_start, key_range_start + NUM_RECORDS / 3);
+
+        sleep(1);
+    }
+
+    /* Clean-up. We do not expect test to reach here. */
+    testutil_check(session->close(session, NULL));
+    testutil_check(conn->close(conn, NULL));
+}
+
+static void
+populate(WT_SESSION *session, int start, int end)
+{
+    WT_CURSOR *cursor_1;
+    WT_CURSOR *cursor_2;
+    time_t t;
+    int i, str_len;
+
+    srand((u_int)time(&t));
+
+    str_len = sizeof(data_str) / sizeof(data_str[0]);
+    for (i = 0; i < str_len - 1; i++)
+        data_str[i] = 'a' + rand() % 26;
+
+    data_str[str_len - 1] = '\0';
+
+    testutil_check(session->open_cursor(session, uri1, NULL, NULL, &cursor_1));
+    testutil_check(session->open_cursor(session, uri2, NULL, NULL, &cursor_2));
+    for (i = start; i < end; i++) {
+        cursor_1->set_key(cursor_1, i);
+        cursor_1->set_value(cursor_1, data_str);
+        testutil_check(cursor_1->insert(cursor_1));
+
+        cursor_2->set_key(cursor_2, i);
+        cursor_2->set_value(cursor_2, data_str);
+        testutil_check(cursor_2->insert(cursor_2));
+    }
+
+    testutil_check(cursor_1->close(cursor_1));
+    testutil_check(cursor_2->close(cursor_2));
+}
+
+static void
+remove_records(WT_SESSION *session, const char *uri, int start, int end)
+{
+    WT_CURSOR *cursor;
+    int i;
+
+    testutil_check(session->open_cursor(session, uri, NULL, NULL, &cursor));
+
+    for (i = start; i < end; i++) {
+        cursor->set_key(cursor, i);
+        testutil_check(cursor->remove(cursor));
+    }
+
+    testutil_check(cursor->close(cursor));
+}
+
+static void
+verify_tables(WT_SESSION *session)
+{
+    WT_CURSOR *cursor_1;
+    WT_CURSOR *cursor_2;
+
+    int ret;
+    int key;
+    int total_keys_1;
+    int total_keys_2;
+
+    char *val_1;
+    char *val_2;
+
+    size_t val_1_size;
+    size_t val_2_size;
+
+    testutil_check(session->open_cursor(session, uri1, NULL, NULL, &cursor_1));
+    testutil_check(session->open_cursor(session, uri2, NULL, NULL, &cursor_2));
+
+    /*
+     * Run over all keys in first table and verify they are present in the second table. Repeat with
+     * all keys from the second table and verify that they are present in the first table;
+     */
+
+    total_keys_1 = 0;
+    while ((ret = cursor_1->next(cursor_1)) == 0) {
```

Comment:
We can avoid code unnecessary duplication by extracting this while loop into a separate method.

Reply:
Good suggestion. I have refactored the code using helper functions in 1512301.